pkg/provider/provider.go

```diff
@@ -72,6 +74,18 @@ func NewProvider(ctx context.Context, adapterLogger logr.Logger, scaleHandler sc
 	}
 	logger = adapterLogger.WithName("provider")
 	logger.Info("starting")
+
+	go func() {
+		if !grpcClient.WaitForConnectionReady(ctx, logger) {
+			grpcClientConnected = false
+			logger.Error(fmt.Errorf("timeout while waiting to establish gRPC connection to KEDA Metrics Service server"), "timeout", "server", grpcClient.GetServerURL())
+		}
```

Comment:
I think that this if case should have a return, otherwise if the connection fails, we will log the error and also the successful connection

Reply:
Good catch, thanks! It should be in the `else`. I refactored this part several times and probably end up with something broken 😄

Reply:
What about just adding the return? We can save one indentation level. Something like:
```golang
	go func() {
		if !grpcClient.WaitForConnectionReady(ctx, logger) {
			grpcClientConnected = false
			logger.Error(fmt.Errorf("timeout while waiting to establish gRPC connection to KEDA Metrics Service server"), "timeout", "server", grpcClient.GetServerURL())
			return
		}
		if !grpcClientConnected {
			grpcClientConnected = true
			logger.Info("Connection to KEDA Metrics Service gRPC server has been successfully established", "server", grpcClient.GetServerURL())
		}
	}()
```